src/transport/SecureSessionMgr.h

```diff
@@ -114,15 +157,23 @@ class DLL_EXPORT SecureSessionMgrDelegate
 class DLL_EXPORT SecureSessionMgr : public Mdns::ResolveDelegate, public TransportMgrDelegate
 {
 public:
+    SecureSessionMgr();
+    ~SecureSessionMgr() override;
+
     /**
      * @brief
-     *   Send a message to a currently connected peer
+     *   Send a message to a currently connected peer.
+     *
+     * @details
+     *   msgBuf contains the data to be transmitted.  If bufferRetainSlot is not null and this function
+     *   returns success, the encrypted data that was sent, as well as various other information needed
+     *   to retransmit it, will be stored in *bufferRetainSlot.
      */
     CHIP_ERROR SendMessage(NodeId peerNodeId, System::PacketBufferHandle msgBuf);
-    CHIP_ERROR SendMessage(PayloadHeader & payloadHeader, NodeId peerNodeId, System::PacketBufferHandle msgBuf);
-    SecureSessionMgr();
-    ~SecureSessionMgr() override;
-
+    CHIP_ERROR SendMessage(PayloadHeader & payloadHeader, NodeId peerNodeId, System::PacketBufferHandle msgBuf,
```

Comment:
Need documentation for these functions. Specifically, what's the use of `bufferRetainSlot`.

Reply:
Already Added:
     * @details
     *   msgBuf contains the data to be transmitted.  If bufferRetainSlot is not null and this function
     *   returns success, the encrypted data that was sent, as well as various other information needed
     *   to retransmit it, will be stored in *bufferRetainSlot.
     */